specification/_spec_utils/WaitForCompletion.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Union of two responses depending on input parameters
+ * and whether or not the server-side action completes
+ * before sending a response. Without setting any parameters
+ * that start with 'wait_for_*' the 'Complete' response is
+ * always returned.
+ */
+export type WaitForCompletion<Complete, Incomplete> = Complete | Incomplete
```

Comment:
Should it be `Delayed` or `Accepted` instead of `Incomplete`?

Actually I'd rather delay this change to a further release, since it needs some special handling in code generators to provide a good developer experience.

For `wait_for_completion` (boolean) I'd like to generate two different endpoints that each hard-code the value of the `wait_for_completion` parameter and return only one side of the union.

Also two endpoints for wait_for_completion_timeout: one with the parameter not set that would return the `Complete` part, and another one with the timeout parameter that would return the union.

I'll capture this in https://github.com/elastic/elasticsearch-specification/issues/1767

Reply:
Oh dang, I didn't mean to include this in this PR, I'll section this off into it's own PR.